edd-auto-register.php

```diff
@@ -461,6 +461,9 @@ private function get_payment( $payment_id, $purchase_data = array() ) {
 			if ( ! empty( $purchase_data['status'] ) ) {
 				$payment->__set( 'status', $purchase_data['status'] );
 			}
+			if ( ! empty( $purchase_data['gateway'] ) ) {
+				$payment->__set( 'gateway', $purchase_data['gateway'] );
```

Comment:
As we did in our previous ones, let's use the direct property set here:
```suggestion
				$payment->gateway = $purchase_data['gateway'];
```